lib/UnrarXLib/volume.cpp

```diff
@@ -20,8 +20,23 @@ bool MergeArchive(Archive &Arc,ComprDataIO *DataIO,bool ShowFileName,char Comman
   Arc.Close();
 
   char NextName[NM];
-  strcpy(NextName,Arc.FileName);
-  NextVolumeName(NextName,(Arc.NewMhd.Flags & MHD_NEWNUMBERING)==0 || Arc.OldFormat);
+  if(DataIO != NULL && (DataIO->m_iSeekTo > Arc.NewLhd.FullPackSize)){
+    if (Arc.Volume && Arc.NotFirstVolume)
+      VolNameToFirstName(Arc.FileName,NextName,(Arc.NewMhd.Flags & MHD_NEWNUMBERING));
+    else
+      strcpy(NextName,Arc.FileName);
+
+    Int64 FastSeek = 0;
+    while(FastSeek+Arc.NewLhd.FullPackSize < DataIO->m_iSeekTo){
+      NextVolumeName(NextName,(Arc.NewMhd.Flags & MHD_NEWNUMBERING)==0 || Arc.OldFormat);
+      FastSeek += Arc.NewLhd.FullPackSize;
```

Comment:
It is possible in theory but never seen it. Was mostly about disclosing the assumption. I didnt implement this before for this reason. Gains are probably worth the possible breakage

-------- Opprinnelig melding --------
Fra: oskvadd <notifications@github.com>
Dato: 10.06.2016 18:33 (GMT+01:00)
Til: xbmc/xbmc <xbmc@noreply.github.com>
Ko: Arne Morten Kvarving <Arne.Morten.Kvarving@sintef.no>, Comment <comment@noreply.github.com>
Emne: Re: [xbmc/xbmc] Speed up startup time for movies in split rar archives. (#9925)


In lib/UnrarXLib/volume.cpp<https://github.com/xbmc/xbmc/pull/9925#discussion_r66642507>:

> @@ -20,8 +20,23 @@ bool MergeArchive(Archive &Arc,ComprDataIO *DataIO,bool ShowFileName,char Comman
>    Arc.Close();
>
>    char NextName[NM];
> -  strcpy(NextName,Arc.FileName);
> -  NextVolumeName(NextName,(Arc.NewMhd.Flags & MHD_NEWNUMBERING)==0 || Arc.OldFormat);
> +  if(DataIO != NULL && (DataIO->m_iSeekTo > Arc.NewLhd.FullPackSize)){
> +    if (Arc.Volume && Arc.NotFirstVolume)
> +      VolNameToFirstName(Arc.FileName,NextName,(Arc.NewMhd.Flags & MHD_NEWNUMBERING));
> +    else
> +      strcpy(NextName,Arc.FileName);
> +
> +    Int64 FastSeek = 0;
> +    while(FastSeek+Arc.NewLhd.FullPackSize < DataIO->m_iSeekTo){
> +      NextVolumeName(NextName,(Arc.NewMhd.Flags & MHD_NEWNUMBERING)==0 || Arc.OldFormat);
> +      FastSeek += Arc.NewLhd.FullPackSize;


Yes, this assumes that the volumes is sized the same(except the last one). I have never seen rar volumes that is not in the same size. You say its possible to make rar volumes without the same size?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/xbmc/xbmc/pull/9925/files/b0e3be09a7418dff76bd96661fa8beda96a4140b#r66642507>, or mute the thread<https://github.com/notifications/unsubscribe/ACC4R8AV0NIgj4lvafOnQQoEjfLJuMTFks5qKZDlgaJpZM4IvGLa>.